src/lib/storage/table.hpp

```diff
@@ -167,6 +167,15 @@ class Table : private Noncopyable {
    */
   size_t estimate_memory_usage() const;
 
+  // Returns whether the table has already been validated (i.e., contains only rows visible to the current transaction)
+  bool is_validated() const;
```

Comment:
Right now it is possible to append chunks WITH mvcc data to a "validated" Table without an assert firing, right?

Reply:
No, because a table can only be validated by the validate operator and once it leaves that operator, it is const. No snappishness intended. ;)